File: app/src/main/java/com/tuvakov/zeyoube/android/dagger/RoomModule.java
package com.tuvakov.zeyoube.android.dagger;

import android.app.Application;

import androidx.room.Room;

import com.tuvakov.zeyoube.android.data.SubscriptionDao;
import com.tuvakov.zeyoube.android.data.VideoDao;
import com.tuvakov.zeyoube.android.data.ZeYouBeDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
abstract class RoomModule {
    @Provides @Singleton
    static ZeYouBeDatabase provideRoomDb(Application application) {
        return Room.databaseBuilder(application, ZeYouBeDatabase.class, "zeyoube_db")
                .fallbackToDestructiveMigration()
                .build();
    }

    @Provides @Singleton
    static SubscriptionDao provideSubscriptionDao(ZeYouBeDatabase zeYouBeDb) {
        return zeYouBeDb.getSubscriptionDao();
    }

    @Provides @Singleton
    static VideoDao provideVideoDao(ZeYouBeDatabase zeYouBeDb) {
        return zeYouBeDb.getVideoDao();
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/dagger/RoomModule.kt
package com.tuvakov.zetube.android.dagger

import android.app.Application
import androidx.room.Room
import com.tuvakov.zetube.android.data.SubscriptionDao
import com.tuvakov.zetube.android.data.VideoDao
import com.tuvakov.zetube.android.data.ZeTubeDatabase
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal object RoomModule {
    @JvmStatic
    @Provides
    @Singleton
    fun provideRoomDb(application: Application): ZeTubeDatabase {
        return Room.databaseBuilder(application, ZeTubeDatabase::class.java, "zetube_db")
                .fallbackToDestructiveMigration()
                .build()
    }

    @JvmStatic
    @Provides
    @Singleton
    fun provideSubscriptionDao(database: ZeTubeDatabase): SubscriptionDao {
        return database.subscriptionDao
    }

    @JvmStatic
    @Provides
    @Singleton
    fun provideVideoDao(database: ZeTubeDatabase): VideoDao {
        return database.videoDao
    }
}